Sources/StreamChat/StateLayer/ChatState.swift

```diff
@@ -96,6 +97,18 @@ public final class ChatState: ObservableObject {
     /// A list of users who are currently typing.
     @Published public private(set) var typingUsers = Set<ChatUser>()
     
+    // MARK: - Watchers
+    
+    /// An array of users who are currently watching the channel.
+    ///
+    /// Use load watchers method in ``Chat`` for populating this array.
+    @Published public private(set) var watchers = StreamCollection<ChatUser>([])
+    
+    /// True, if all the watchers were loaded with paginated loading.
+    ///
+    /// The value is set to true when the pagination request returns less than the pagination page size.
+    @Published public private(set) var hasLoadedAllWatchers = false
```

Comment:
Debatable if we should provide this 🤔

Reply:
Let me take it away since I also felt like it was too much. If our UI code would benefit from this, we can always add it or add something similar on the UI level instead.